### PR TITLE
Create Version 1

### PR DESCRIPTION
**Pull request by Auto19**
Initial version of the AD Hunt tool

### CODE REVIEW

File: adhunt.py [hunk 1]
```diff
@@ -0,0 +1,617 @@
...
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+    BOLD = '\033[1m'
+    UNDERLINE = '\033[4m'
+    INSTALL = '\x1B[38;5;166m'
+
+if(os.name != "posix"):
+	print("AD Hunt only works in a linux enviroment.")
+
```
Comment (robert-todora) [inline adhunt.py:37]: Should you exit here?

Comment (Auto19) [reply adhunt.py:38]: yes

File: adhunt.py [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+	if(not c.bind()):
+		print(c.result)
+		sys.exit(1)
+	print(s.info)
+	
+	sys.exit(0)
+
+if(not args.domain):
+	print("A domain must be supplied.")
+
```
Comment (robert-todora) [inline adhunt.py:94]: exit here as well?

File: adhunt.py [hunk 3]
```diff
@@ -0,0 +1,617 @@
...
+	print(f"{bcolors.INSTALL}[+] Installing NMAP{bcolors.ENDC}")
+	os.system("apt install nmap")
+	sys.exit(0)
+
+
+# TODO most LDAP searchs will fetch deactivated accounts
+
+if(not args.domain_controller_ip):
+	print("Must specify the ip of a domain controller with -dc-ip")
+
```
Comment (robert-todora) [inline adhunt.py:73]: exit here?

File: adhunt.py [hunk 4]
```diff
@@ -0,0 +1,617 @@
...
+if(args.username != None and args.password == None):
+	print("If a username is supplied a password must also be supplied")
+
```
Comment (robert-todora) [inline adhunt.py:77]: exit here?

**Review by robert-todora — APPROVED** [review]
Maybe add a requirements.txt for the python packages they'll need to run the project, like ldap3, dns.resolver, impacket, etc. 

Also, this might be better to do for v2 but it would make it more readable to put each component of this script in its own function definition and then have a main function that'll run call each function sequentially, and that way it'll be easier to handle errors with try except blocks and stuff

**Pull request merged**